src/librustc_mir/transform/check_consts/ops.rs

```diff
@@ -268,18 +268,24 @@ impl NonConstOp for Panic {
 #[derive(Debug)]
 pub struct RawPtrComparison;
 impl NonConstOp for RawPtrComparison {
-    fn feature_gate() -> Option<Symbol> {
-        Some(sym::const_compare_raw_pointers)
-    }
-
     fn emit_error(&self, ccx: &ConstCx<'_, '_>, span: Span) {
-        feature_err(
-            &ccx.tcx.sess.parse_sess,
-            sym::const_compare_raw_pointers,
+        let mut err = ccx.tcx.sess.struct_span_err(
             span,
-            &format!("comparing raw pointers inside {}", ccx.const_kind()),
-        )
-        .emit();
+            "pointers cannot be compared in a meaningful way during const eval.",
+        );
+        err.note(
+            "It is conceptually impossible for const eval to know in all cases whether two \
+             pointers are equal. While sometimes it is clear (the address of a static item \
+             is never equal to the address of another static item), comparing an integer \
+             address with any allocation's address is impossible to do at compile-time.",
+        );
+        err.note(
+            "That said, there's the `ptr_guaranteed_eq` intrinsic which returns `false` for all \
+             comparisons where CTFE isn't sure whether two addresses are equal. The mirror \
+             intrinsic `ptr_guaranteed_eq` returns `false` for all comparisons where CTFE isn't \
```

Comment:
```suggestion
             intrinsic `ptr_guaranteed_ne` returns `false` for all comparisons where CTFE isn't \
```
But also your explanation is using double-negation ("false if not sure"), wouldn't it be easier to describe this as "true if it will definitely be equal, and false if we cannot be sure"?

Reply:
I improved the explanation (and got rid of the double negative) lmk what you think